pontoon/terminology/models.py

```diff
@@ -62,6 +101,143 @@ def translation(self, locale):
             except (AttributeError, TermTranslation.DoesNotExist):
                 return None
 
+    @property
+    def localizable(self):
+        """
+        Check if the term is localizable.
+        """
+        if self.do_not_translate:
+            return False
+
+        if self.forbidden:
+            return False
+
+        if self.definition == "":
+            return False
+
+        return True
+
+    def entity_comment(self):
+        """
+        Generate entity comment from the term.
+        """
+        comment = "{}. {}.".format(
+            self.part_of_speech.capitalize(), self.definition.capitalize().rstrip("."),
+        )
+
+        if self.usage:
+            comment += " E.g. {}.".format(self.usage.capitalize().rstrip("."))
+
+        return comment
+
+    def create_entity(self):
+        """
+        An Entity must be created (or deobsoleted) for a Term according to the
+        following rules:
+        - Entity.string contains content of Term.text.
+        - Entity.comment contains joint content of several fields:
+          Term.part_of_speech. Term.definition. E.g.: Term.usage.
+        """
+        resource = Resource.objects.get(project__slug="terminology")
+
+        entity, created = Entity.objects.get_or_create(
+            string=self.text, comment=self.entity_comment(), resource=resource,
+        )
+
+        # Using update() to avoid circular Term.save() call
+        Term.objects.filter(pk=self.pk).update(entity_id=entity.id)
+
+        if created:
+            entity.order = (
+                Entity.objects.filter(resource=resource).order_by("-order")[0].order
```

Comment:
[question] What happens with `entity.order` when you call create_entity two times in a row? I think it will stay the same for the new entities.

Reply:
Hmmmm, right. That should be +1.